macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,280 @@
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample's encoding option
+      - id: base_frequency
+        type: u1
+        doc: base frequency of sample
+        -orig-id: baseFrequency
+      - id: num_frames
+        type: u4
+      - id: aiff_sample_rate
+        size: 10
+        doc: rate of original sample (Extended80)
+      - id: marker_chunk
+        -orig-id: markerChunk
+        type: u4
+        doc: reserved
+      - id: instrument_chunks
+        -orig-id: instrumentChunks
+        type: u4
+        doc: pointer to instrument info
+      - id: aes_recording
+        -orig-id: AESRecording
+        type: u4
+        doc: pointer to audio info
+      - id: sample_size
+        -orig-id: sampleSize
+        type: u2
+        doc: number of bits per sample
+      - id: reserved
+        -orig-id: futureUse1, futureUse2, futureUse3, futureUse4
+        size: 14
+        doc: reserved
+      - id: sample_area
+        -orig-id: sampleArea
+        size: num_frames * num_channels * sample_size / 8
+        doc: sampled-sound data
+  compressed_sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or 0 if samples follow data structure)
+      - id: num_channels
+        -orig-id: numChannels
+        type: u4
+        doc: number of channels in sample
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: unsigned_fixed_point
+        doc: The rate at which the sample was originally recorded.
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample's encoding option
+      - id: base_frequency
+        type: u1
+        doc: base frequency of sample
+        -orig-id: baseFrequency
+      - id: num_frames
+        type: u4
+      - id: aiff_sample_rate
+        size: 10
+        doc: rate of original sample (Extended80)
+      - id: marker_chunk
+        -orig-id: markerChunk
+        type: u4
+        doc: reserved
+      - id: format
+        size: 4
+        type: str
+        encoding: ASCII
+        doc: data format type
+      - id: reserved
+        -orig-id: futureUse2
+        size: 4
+        doc: reserved
+      - id: state_vars_ptr
+        -orig-id: stateVars
+        type: u4
+        doc: pointer to StateBlock
+      - id: left_over_samples_ptr
+        -orig-id: leftOverSamples
+        type: u4
+        doc: pointer to LeftOverBlock
+      - id: compression_id
+        -orig-id: compressionID
+        type: u2
+        doc: ID of compression algorithm
```

Comment:
I agree, that it needs to be a signed value. There are some reserved IDs, but in principle every value is valid, so I don't think an enum would be feasible.

Reply:
> There are some reserved IDs, but in principle every value is valid, so I don't think an enum would be feasible.

Well, if you think so, you can at least define both integer and enum variant of this attribute:

```ksy
seq:
  - id: compression_id
    type: u2
instances:
   compression_type:
    value: compression_id
    enum: compression_type_enum
enums:
  compression_type_enum:
    -2: variableCompression
    -1: fixedCompression
    0: notCompressed 
    3: threeToOne
    4: threeToOne
```

It's definitely better than just an opaque integer.

A user of this can first switch on the enum values of `compression_type` and if it has an unknown enum value, then access the raw integer `compression_id` and do something with it.

Reply:
Thats a good idea! Will do that.

Reply:
added the enum